lib/ts/recipe/totp/components/features/mfa/index.tsx

```diff
@@ -0,0 +1,398 @@
+        [userContext]
+    );
+
+    const handleLoadError = React.useCallback(
+        () => dispatch({ type: "setError", showAccessDenied: true, error: "SOMETHING_WENT_WRONG_ERROR" }),
+        [dispatch]
+    );
+    const onLoad = React.useCallback(
+        async (mfaInfo: Awaited<ReturnType<typeof fetchMFAInfo>>) => {
+            let error: string | undefined = undefined;
+            const errorQueryParam = getQueryParams("error");
+            const doSetup = getQueryParams("setup");
+            if (errorQueryParam !== null) {
+                error = "SOMETHING_WENT_WRONG_ERROR";
+            }
+            const isAlreadySetup = mfaInfo.factors.isAlreadySetup.includes("totp");
+
+            // If we have finished logging in and the user was redirected here
+            // otherwise the user can use the "choose another factor" button
+            const showBackButton = mfaInfo.factors.next.length === 0;
+            const showFactorChooserButton = mfaInfo.factors.next.length > 1;
+
+            let deviceInfo: TOTPDeviceInfo | undefined;
+            if (doSetup || !isAlreadySetup) {
+                let createResp;
+                try {
+                    dispatch({
+                        type: "load",
+                        deviceInfo: undefined,
+                        error,
+                        showBackButton,
+                        showFactorChooserButton,
+                        showAccessDenied: false,
+                        callingCreateDevice: true,
+                    });
+                    createResp = await recipeImpl.createDevice({ userContext });
+                } catch {
+                    dispatch({
+                        type: "setError",
+                        showAccessDenied: true,
+                        error: "SOMETHING_WENT_WRONG_ERROR",
+                    });
+                    return;
+                }
+                if (createResp.status !== "OK") {
+                    dispatch({
+                        type: "setError",
+                        showAccessDenied: true,
+                        error: "SOMETHING_WENT_WRONG_ERROR",
+                    });
+                    return;
+                }
+                deviceInfo = {
+                    ...createResp,
+                };
+                delete (deviceInfo as any).status;
+            }
+
+            // No need to check if the component is unmounting, since this has no effect then.
+            dispatch({
+                type: "load",
+                deviceInfo,
+                error,
+                showBackButton,
+                showFactorChooserButton,
+                showAccessDenied: false,
+                callingCreateDevice: false,
+            });
+        },
+        [dispatch, recipeImpl, userContext]
+    );
+
+    useOnMountAPICall(fetchMFAInfo, onLoad, handleLoadError);
+}
+
```

Comment:
if the user has reached the limit, and they refresh the page, what does the user see? Do they see the limit reached UI again or enter TOTP which would fail again?

Reply:
They'd see the enter TOTP screen, which'd fail if enough time hasn't passed